opsdroid/connector/matrix/connector.py

```diff
@@ -281,8 +281,8 @@ async def _parse_sync_response(self, response):
                             _LOGGER.error(
                                 f"Failed to decrypt event {event}"
                             )  # pragma: nocover
-                        return await self._event_creator.create_event(
```

Comment:
@SolarDrew this return causes the loop to exit on the first event it encounters. By not returning and calling `opsdroid.parse` directly in the loop we don't terminate the loop early.

Reply:
Ah ok, that makes sense. And I guess that's what I get for only looking at the diff.

Reply:
I don't see why, but if there was some reason to not call parse in this context, you could also year this as a generator and yield the message to parse, returning when the loop exits.

Reply:
That might work quite well.

Reply:
Yeah that's an idea. @jacobtomlinson how would that mesh with your views on beginner-friendliness of the codebase?

Reply:
I have implemented this, it's nice.